images/macos/provision/core/postgresql.sh

```diff
@@ -1,18 +1,26 @@
-#!/bin/bash
+#!/bin/bash -e -o pipefail
 
 #Install latest version of postgresql
 brew install postgres
 
 #Service postgresql should be started before use.
 brew services start postgresql
 
-#Verify that PostgreSQL is ready for accept incoming connections.
-# exit codes:
-# ready - 0
-# reject - 1
-# connection timeout - 2
-# incorrect credentials or parameters - 3
-pg_isready
+#Verify PostgreSQL is ready for accept incoming connections
+echo "Check PostgreSQL service is running"
+i=10
+COMMAND='pg_isready'
+while [ $i -gt 0 ]; do
+    echo "Check PostgreSQL service status"
+    eval $COMMAND && break
+    ((i--))
+    if [ $i == 0 ]; then
+        echo "PostgreSQL service not ready, all attempts exhausted"
+        exit 1
+    fi
+    echo "PostgreSQL service not ready, wait 10 more sec, attempts left: $i"
+    sleep 10
+done
```

Comment:
Do we really need this validation in installation script?
Can we move test to tests files?

PowerShell command `Start-Service` can wait until service is started fully

Reply:
The check was initially implemented with some intent for it 😃
Does it make sense to do a test if we need postgre being stopped at the end of the build anyway?

Reply:
I think we need this script as an indicator that postgres was installed correctly